src/Numerics/SoaSphericalTensor.h

```diff
@@ -169,6 +211,109 @@ inline SoaSphericalTensor<T>::SoaSphericalTensor(const int l_max, bool addsign)
     }
 }
 
+PRAGMA_OFFLOAD("omp declare target")
+template<typename T>
+inline void SoaSphericalTensor<T>::evaluate_bare_impl(const T x,
```

Comment:
Could you make `evaluate_bare` calling `evaluate_bare_impl` to minimize code duplication?

Reply:
I'm not sure how to handle it for SoaSphericalTensor.
[`evaluateVGL` calls `evaluate_bare`](https://github.com/kgasperich/qmcpack/blob/5821946cf2a6fff89c589c21fbbb66e1a84af9b7/src/Numerics/SoaSphericalTensor.h#L415-L416) and passes the scratch space of the object, but `evaluate_bare_impl` is static, so it shouldn't modify `cYlm`.
This is what I get when I try to have `evaluate_bare` call `evaluate_bare_impl`
```
27: AddressSanitizer:DEADLYSIGNAL
27: =================================================================
27: ==1264599==ERROR: AddressSanitizer: SEGV on unknown address 0x03e800134bd7 (pc 0x561133b7a46b bp 0x7ffff530a780 sp 0x7ffff530a720 T0)
27: ==1264599==The signal is caused by a READ memory access.
27:     #0 0x561133b7a46b in qmcplusplus::SoaSphericalTensor<double>::evaluate_bare_impl(double, double, double, double*, unsigned long, double const*, double const*) (/home/keving/work/qmcpack/qmcpack-ao-resource/build_asan/src/QMCWaveFunctions/tests/test_wavefunction_sposet+0x17bf46b) (BuildId: 34e365f3ed46d03bce9509e4c7975c5a04bb5060)
27:     #1 0x561133b7d9a9 in qmcplusplus::SoaSphericalTensor<double>::evaluateVGL(double, double, double) (/home/keving/work/qmcpack/qmcpack-ao-resource/build_asan/src/QMCWaveFunctions/tests/test_wavefunction_sposet+0x17c29a9) (BuildId: 34e365f3ed46d03bce9509e4c7975c5a04bb5060)
```

I think there's a similar issue with [`SoaCartesianTensor<T>::evaluateV(T, T, T)` ](https://github.com/kgasperich/qmcpack/blob/5821946cf2a6fff89c589c21fbbb66e1a84af9b7/src/Numerics/SoaCartesianTensor.h#L66-L83)

Reply:
I would've thought that the static function could modify the data since it was being passed as an argument rather than accessing it via `this`, so maybe the error comes from something else?

The change I made to get that error was to just have `evaluate_bare` call `evaluate_bare_impl`

```
template<typename T>
inline void SoaSphericalTensor<T>::evaluate_bare(T x, T y, T z, T* restrict Ylm) const
{
  evaluate_bare_impl(x, y, z, Ylm, Lmax, FactorL.data(), FactorLM.data());
}
```

maybe it's a `const` related issue?